### PR TITLE
Use ExternalData and remove some scripts for testing

### PR DESCRIPTION
**Pull request by jmcarcell**
BEGINRELEASENOTES
- Use ExternalData and remove some scripts for testing to fix running tests in parallel

ENDRELEASENOTES

Tests are failing sometimes; there are different scripts trying to download the same files and there are some race conditions. With these changes the files are downloaded at configure time as we do with other repositories.

### CODE REVIEW

**Comment by tmadlener** [comment]
The `clicRec` test still seems to try to get his data from somewhere. At least the test output suggests that.

File: test/CMakeLists.txt [hunk 1]
```diff
@@ -35,82 +41,79 @@ add_library(MarlinTestProcessors SHARED src/PseudoRecoProcessor.cc src/TrivialMC
...
+add_test( clicRec bash -c ${CMAKE_CURRENT_SOURCE_DIR}/scripts/clicRec.sh )
+
+# Test converter constants
+add_test( converter_constants bash -c ${CMAKE_CURRENT_SOURCE_DIR}/scripts/converter_constants.sh )
+
+# Test indicating -1 to go over all events
+ExternalData_Add_Test( marlinwrapper_tests NAME all_events_bounds COMMAND ${K4RUN} ${CMAKE_CURRENT_SOURCE_DIR}/gaudi_opts/simple_processors.py --num-events=1 --LcioEvent.Files DATA{${PROJECT_SOURCE_DIR}/test/input_files/muons.slcio})
+# set_tests_properties ( all_events_bounds
+#   PROPERTIES
+#     PASS_REGULAR_EXPRESSION "Application Manager Terminated successfully with a user requested ScheduledStop")
```
Comment (tmadlener) [inline test/CMakeLists.txt:65]:
```suggestion
```

File: test/gaudi_opts/test_global_converter_maps.py [hunk 2]
```diff
@@ -18,9 +18,9 @@
 # limitations under the License.
 #
 
-import os
 
-from Gaudi.Configuration import *
+from Gaudi.Configuration import INFO, DEBUG
+import sys
```
Comment (tmadlener) [inline test/gaudi_opts/test_global_converter_maps.py:23]:
```suggestion
import os
```

`sys` should not be necessary, but `os` should?

Comment (jmcarcell) [reply test/gaudi_opts/test_global_converter_maps.py:23]: Neither are needed

**Comment by jmcarcell** [comment]
I'm not sure now why one of the tests that I didn't touch fails, testing them on Alma 9 they pass...

**Comment by tmadlener** [comment]
Could it be an issue with the file that we download? It looks a bit like it just can't open the file. Maybe FetchContent stores it in a different place? https://github.com/key4hep/k4MarlinWrapper/actions/runs/7829896766/job/21364955222?pr=167#step:3:3752

**Comment by jmcarcell** [comment]
It was just a wrong path. For some reason running the tests on the alma9 image used to build without cvmfs worked fine... but when running on the image the CI is using then I could reproduce it. CentOS 7 is broken but the clic reconstruction is broken there: https://github.com/key4hep/key4hep-spack/issues/558

**Review by tmadlener — APPROVED** [review]

**Pull request merged**